pkgs/by-name/or/orca-slicer/package.nix

```
@@ -15,6 +17,18 @@ bambu-studio.overrideAttrs (
       hash = "sha256-h+cHWhrp894KEbb3ic2N4fNTn13WlOSYoMsaof0RvRI=";
     };
 
+    nativeBuildInputs = (previousAttrs.nativeBuildInputs or [ ]) ++ [
+      pkg-config
+    ];
+
+    buildInputs = (previousAttrs.buildInputs or [ ]) ++ [
```

Let's remove webkitgtk_4_0 to reduce the build closure.


```suggestion
    buildInputs = (lib.remove webkitgtk_4_0 previousAttrs.buildInputs or [ ]) ++ [
```

---

Have try to apply your suggestion but that not help.

---

The reason the build fails is that the patch needs to be updated to link against webkit2gtk-4.1.

I was able to build orca-slicer with the following patch:

```patch
From 414dfb6cef6cb144625982a02296bf3086e8431e Mon Sep 17 00:00:00 2001
From: Sefa Eyeoglu <contact@scrumplex.net>
Date: Wed, 27 Nov 2024 19:53:32 +0100
Subject: [PATCH] orca-slicer: link against webkitgtk-4.1

Signed-off-by: Sefa Eyeoglu <contact@scrumplex.net>
---
 pkgs/by-name/or/orca-slicer/package.nix                        | 3 ++-
 ...-not-for-upstream-CMakeLists-Link-against-webkit2gtk-.patch | 2 +-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/pkgs/by-name/or/orca-slicer/package.nix b/pkgs/by-name/or/orca-slicer/package.nix
index 00c3cdea2938..9ff5e6a1e490 100644
--- a/pkgs/by-name/or/orca-slicer/package.nix
+++ b/pkgs/by-name/or/orca-slicer/package.nix
@@ -2,6 +2,7 @@
   lib,
   fetchFromGitHub,
   bambu-studio,
+  webkitgtk_4_0,
   webkitgtk_4_1,
   pkg-config,
 }:
@@ -21,7 +22,7 @@ bambu-studio.overrideAttrs (
       pkg-config
     ];
 
-    buildInputs = (previousAttrs.buildInputs or [ ]) ++ [
+    buildInputs = (lib.remove webkitgtk_4_0 previousAttrs.buildInputs or [ ]) ++ [
       webkitgtk_4_1
     ];
 
diff --git a/pkgs/by-name/or/orca-slicer/patches/0001-not-for-upstream-CMakeLists-Link-against-webkit2gtk-.patch b/pkgs/by-name/or/orca-slicer/patches/0001-not-for-upstream-CMakeLists-Link-against-webkit2gtk-.patch
index 15f1bf8f0b59..4497271938b5 100644
--- a/pkgs/by-name/or/orca-slicer/patches/0001-not-for-upstream-CMakeLists-Link-against-webkit2gtk-.patch
+++ b/pkgs/by-name/or/orca-slicer/patches/0001-not-for-upstream-CMakeLists-Link-against-webkit2gtk-.patch
@@ -23,7 +23,7 @@ index 9c5cb96..e92a0e3 100644
  
 +# We link against webkit2gtk symbols in src/slic3r/GUI/Widgets/WebView.cpp
 +if (CMAKE_SYSTEM_NAME STREQUAL "Linux")
-+    target_link_libraries(libslic3r_gui "-lwebkit2gtk-4.0")
++    target_link_libraries(libslic3r_gui "-lwebkit2gtk-4.1")
 +endif ()
 +
  # Link the resources dir to where Slic3r GUI expects it
-- 
2.47.0
```